### PR TITLE
Add typing for LocalizeJS Library

### PR DESCRIPTION
**Pull request by salbahra**
This adds the library typings for LocalizeJS Library script loaded using the web snippet which makes a global variable available named `Localize`. This differs from the `localizejs` package which is for node specifically (and is deprecated) whereas this caters to the browser script as documented here: https://help.localizejs.com/docs/library-api.

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/localizejs/index.d.ts
Comment (plantain-00) [inline types/localizejs/index.d.ts]: If I'm right, the name should be same with the package name.

File: types/localizejs/index.d.ts
Comment (plantain-00) [inline types/localizejs/index.d.ts]: I didn't find any API related to the `initialize` at https://www.npmjs.com/package/localizejs , is it same package?

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/localizejs/index.d.ts
Comment (salbahra) [reply types/localizejs/index.d.ts]: No this is not the same package, sorry for the confusion. I tried to outline this in the header of the request. The end service is the same however the NPM package appears to be specific to their node implementation however we need this for an application using the web browser snippet (similar to how Google Analytics is added). Should I use a different name to avoid the confusion and if so do you have a recommendation?

File: types/localizejs/index.d.ts
Comment (salbahra) [reply types/localizejs/index.d.ts]: There is no package available for this library on NPM. It is added using a snippet and exposes a global variable. Would you suggest I use a different name?

File: types/localizejs/index.d.ts
Comment (plantain-00) [reply types/localizejs/index.d.ts]: I'm not sure, let's ask @andy-ms

Comment (salbahra) [reply types/localizejs/index.d.ts]: Just to point out, the package on Github point out it is deprecated and defer to the web API:

```
This module has been deprecated

Please use the official Localize API.
```

Since they have no other library other than the one offered by the web snippet do you think it is safe continuing to use this name? The version on the current package is 0.4.2 and I have used 1.0 here which is greater.

Do you think knowing this helps? I am needing this for a current project and would like to stop using the local copy. I am correct in assuming the type can't be published to npm before merging into this repository?

Thanks!

Comment (plantain-00) [reply types/localizejs/index.d.ts]: I'm not sure, according to the document
>  If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.

You may have to change the name to avoid conflict with existing package.
This is a weekend, so it seems no mergers online now, so you may have to wait a few days.

@alloy @johnnyreilly Can you give suggestions for the no npm package case?

Comment (alloy) [reply types/localizejs/index.d.ts]: Sorry, this is out of my wheelhouse.

Comment (salbahra) [reply types/localizejs/index.d.ts]: I noticed they called this the Library API throughout their documentation and links so I have renamed this to `LocalizeJS Library` and `localizejs-library` which no package exists for. Let me know if this helps.

Thank you!

**Review by plantain-00 — APPROVED** [review]

File: types/localizejs/index.d.ts
Comment (andy-ms) [reply types/localizejs/index.d.ts]: Yes, use another name -- `localizejs-library` is good.

**Pull request merged**
